### PR TITLE
#162948835 Document API using Swagger

### PR DESCRIPTION
**Pull request by KMaina**
#### What does this PR do?
Enables Swagger API documentation
#### Description of Task to be completed?

- Update the `requirements.txt`
- Add configurations in `settings.py`
- Add a route in the `urls.py`

#### How should this be manually tested?

- Ensure that the app is running
- Navigate to the root of the app (http://127.0.0.1:8000/)
- View the documentation

#### What are the relevant pivotal tracker stories?
[#162948835](https://www.pivotaltracker.com/story/show/162948835)
#### Screenshots
Login documentation
login
Registration documentation
register

### CODE REVIEW

File: authors/settings.py [hunk 1]
```diff
@@ -146,3 +149,21 @@
     #     'authors.apps.authentication.backends.JWTAuthentication',
```
Comment (kwameasiago) [inline authors/settings.py]: Do not push commented code

File: authors/settings.py [hunk 2]
```diff
@@ -146,3 +149,21 @@
...
+        'delete',
+    ],
+    'SECURITY_DEFINITIONS': {
+        'api_key': {
+            'type': 'apiKey',
+            'name': 'Authorization',
+            'in': 'header'
+        },
+    },
+}
```
Comment (kwameasiago) [inline authors/settings.py]: Add a line at the end of every file

File: authors/settings.py (hunk 1 above)
Comment (KMaina) [reply authors/settings.py]: Noted

File: authors/settings.py (hunk 2 above)
Comment (KMaina) [reply authors/settings.py]: Noted

File: authors/settings.py (hunk 1 above)
Comment (andrewhingah) [reply authors/settings.py]: You can also include a screenshot of the documentation

**Review by kwameasiago — APPROVED** [review]
Looks good

File: requirements.txt [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
 django-extensions==2.1.4
+django-rest-swagger==2.1.0
 djangorestframework==3.9.0
-PyJWT==1.7.1
-six==1.12.0 
+idna==2.8
+itypes==1.1.0
+Jinja2==2.10
+MarkupSafe==1.1.0
+openapi-codec==1.3.2
```
Comment (ignatiusukwuoma) [inline requirements.txt]: Do we need all these new packages you added?

**Review by andrewhingah — APPROVED** [review]
Looks good. Only that the travis build is failing.

File: requirements.txt (hunk 3 above)
Comment (KMaina) [reply requirements.txt]: Let me have a look at them

**Comment by ignatiusukwuoma** [comment]
I think you have unnecessary packages installed in your virtual environment. Uninstall all packages that are not in previous code and are not required for the swagger docs.

File: requirements.txt [hunk 4]
```diff
@@ -1,13 +1,12 @@
 Django==2.1.4
 django-cors-middleware==1.3.1
 django-extensions==2.1.4
+django-rest-swagger==2.1.0
 djangorestframework==3.9.0
-PyJWT==1.7.1
-six==1.12.0
```
Comment (ignatiusukwuoma) [inline requirements.txt:6]: You shouldn't remove packages from this file since it's not your PR that added them.

File: requirements.txt [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
 django-cors-middleware==1.3.1
 django-extensions==2.1.4
+django-rest-swagger==2.1.0
 djangorestframework==3.9.0
-PyJWT==1.7.1
-six==1.12.0 
 psycopg2-binary==2.7.6.1
 pycodestyle==2.4.0
 dj-database-url==0.5.0
-pytz==2018.9
```
Comment (ignatiusukwuoma) [inline requirements.txt:10]: ditto

(means I have already the feedback above)

File: requirements.txt (hunk 4 above)
Comment (KMaina) [reply requirements.txt:6]: From the branch I see that PyJWT is still in the requirements. As for the others I will add them

File: requirements.txt (hunk 5 above)
Comment (KMaina) [reply requirements.txt:10]: Done

**Review by ignatiusukwuoma — APPROVED** [review]

**Pull request merged**